UI/UISearchBar.cs

```diff
@@ -141,6 +150,48 @@ private void HandleTextInput()
 			}
 		}
 
+		private void HandleRight()
+		{
+			if (cursorPosition != Text.Length)
+			{
+				//Check if first character is Whitespace and skip if so
+				if (char.IsWhiteSpace(Text[cursorPosition]))
+				{
+					cursorPosition += 1;
+				}
+
+				var newPos = Text.IndexOf(' ', cursorPosition);
+
+				if (newPos == -1)
+				{
+					cursorPosition = Text.Length - 1;
+				}
+				else
+				{
+					cursorPosition = newPos - 1;
+				}
+			}
+		}
+
+		private void HandleLeft()
+		{
+			if (cursorPosition > 0)
+			{
+				var newPos = Text.Substring(0, cursorPosition).TrimEnd().LastIndexOf(' ');
```

Comment:
This can be optimized by a negligible amount by using
`var newPos = Text.AsSpan(0, cursorPosition).TrimEnd().LastIndexOf(' ');`

Reply:
Interesting. Didn't know.